arm/Microsoft.Compute/virtualMachines/.bicep/nested_networkInterface.bicep

```
@@ -73,6 +73,8 @@ resource networkInterface 'Microsoft.Network/networkInterfaces@2021-03-01' = {
         subnet: {
           id: ipConfiguration.subnetId
         }
+        loadBalancerBackendAddressPools: contains(ipConfiguration, 'loadBalancerBackendAddressPools') ? ipConfiguration.loadBalancerBackendAddressPools : null
```

I'd suggest to move the complexity of building the loadBalancerBackendAddressPools object here to simplify the usage and avoid asking for the "id" property in the parameter file, similarly to what done for the subnet above
```suggestion
        loadBalancerBackendAddressPools: contains(ipConfiguration, 'loadBalancerBackendAddressPools') ? { 
        "id": ipConfiguration.loadBalancerBackendAddressPools
        } : null
```

---

@eriqua : What do we do here? Ref our discussion, its a var with nested loops if we want to do this. And that might be difficult to achieve. Thoughts?

---

Agreed. If it wouldn't be nested I'd agree